packages/core/src/dialects/abstract/query-interface.d.ts

```diff
@@ -218,6 +218,11 @@ export interface AddForeignKeyConstraintOptions extends BaseConstraintOptions {
   references?: {
     table: TableName,
     field: string,
+    fields: never,
+  } | {
+    table: TableName,
+    field: never,
```

Comment:
Is this never property necessary?

Reply:
From what I could find, that's the only way we can XOR - I suppose we could have
```
{ table: TableName, fields: string[] } | { table: TableName, field: string }
```
if you'd prefer

Reply:
Good call - it actually broke the typings in the TS test. Removed in the upcoming commit